prepping for CAP 1.5.2

gaktive opened the pull request.

gaktive committed:
prepping for CAP 1.5.2

Commit:
Updated with 1.5.2

Commit:
Updated year to 2020

gaktive committed:
updated component versions for 1.5.2

referred to role-manifest.yml in `rc` branch of scf for 2.20.1

gaktive commented:
Last key piece are the known issues. Let's see if QA blesses & sanctifies scf 2.20.1 as releaseworthy.

btat commented on adoc/Release-Notes.adoc:
```diff
@@ -33,6 +33,73 @@ endif::[]
+* Advised operator of the need for CSR approval for Eirini internal registry in Helm notes
+* Pod restarts for many roles during UAA and SCF startup lessened for faster deployment
+* Expanded GoRouter ciphersuite to be compatible with a wider range of clients
+* Made helm chart compatible with {k8s} 1.16
```
```suggestion
* Made Helm chart compatible with {k8s} 1.16
```

btat reviewed (commented):
One small typo, otherwise LGTM

gaktive committed:
Fixed Helm spelling

Co-Authored-By: Billy Tat <btat@suse.com>

gaktive committed:
initial write-up for Minibroker & Kubernetes 1.16

gaktive committed:
downgraded loggregator version & TLS additions

Based on what's anticpated for scf 2.20.2, started fleshing out CAP 1.5.2 known issues.

gaktive committed:
changed month

btat commented on adoc/Release-Notes.adoc:
```diff
@@ -99,6 +108,17 @@ endif::[]
+
+* If you are upgrading {product} to 1.5.2 and already use Minibroker to connect to external databases and are using {k8s} 1.16 or higher, which is the case with {caasp} 4.1, you will need to update the database version to a compatible version and migrate your data over via the database's suggested mechanism. This may require a database export/import.
+
+* Starting with {product} 1.5.2, you no longer need to set `UAA_CA_CERT` when using an external UAA with a certificate signed by a well known Certificate Authority (CA). It is only needed when you use an external UAA with either a certificate generated by the `secret-generator` a self-signed certificate.
```
```suggestion
* Starting with {product} 1.5.2, you no longer need to set `UAA_CA_CERT` when using an external UAA with a certificate signed by a well known Certificate Authority (CA). It is only needed when you use an external UAA with either a certificate generated by the `secret-generator` or a self-signed certificate.
```

btat commented on adoc/Release-Notes.adoc:
```diff
@@ -34,14 +34,23 @@ endif::[]
 ** Optimized startup time for various roles
 ** Added/fixed podAntiAffinity rules for various roles
+* The Stratos UI has been updated to version 2.7.0:
+** Tech preview for helm feature
```
```suggestion
** Tech preview for Helm feature
```

gaktive replied:
this line will be junked once I review the actual 2.7.0 changelog so mark this as incomplete for now ;)

btat reviewed (commented):
Found a couple minor typos

gaktive committed:
fixed grammar

Co-Authored-By: Billy Tat <btat@suse.com>

gaktive committed:
preliminary info around stratos console & metrics for 1.5.2

gaktive committed:
known issue for USB on Azure

gaktive committed:
updated nfs-volume-release based on RC4

satadruroy committed:
More release note changes for external database

satadruroy committed:
Merge branch 'cap-152' of https://github.com/SUSE/cap-release-notes into cap-152

btat reviewed (commented):
:+1:  to update from @satadruroy

satadruroy committed:
Minor cleanups and re-organization

gaktive commented on adoc/Release-Notes.adoc:
```diff
@@ -119,7 +119,7 @@ Note that the external database server *must* support both TLS and non-encrypted
 
 * {product} 1.5.2 has been tested successfully with external database instances hosted on Amazon RDS MariaDB and Azure MariaDB. Amazon Aurora, Amazon RDS MySQL or Azure MYSQL are not yet supported configurations.  
```
Should we account for the 2nd sentence as to why? I'd lean towards just stating what we successfully tested in the release notes.

satadruroy replied:
Yeah, we can drop the 2nd sentence. Should we add "Any other external database setup is not supported at this point."?

gaktive replied:
2nd sentence removed. Awaiting more feedback from @troytop

gaktive committed:
added external db upgrade caveat

jandubois commented on adoc/Release-Notes.adoc:
```diff
@@ -33,6 +33,109 @@ endif::[]
+
+* Starting with {product} 1.5.2, you no longer need to set `UAA_CA_CERT` when using an external UAA with a certificate signed by a well known Certificate Authority (CA). It is only needed when you use an external UAA with either a certificate generated by the `secret-generator` or a self-signed certificate.
+
+* Prior to {product} 1.5.2, it's possible that rotating the CCDB secret keys may have been broken. This has been fixed with this version: please consult https://documentation.suse.com/suse-cap/1.5.2/html/cap-guides/cha-cap-ccdb-secret-rotation.html 
```
It was just the documentation, that was broken, not the rotation itself. And it is not "possible", the documentation was definitely broken, with the single exception of an empty database, in which case the instructions would have worked (because there would be nothing to decrypt/re-encrypt).

gaktive replied:
Updated accordingly to indicate the documentation has been corrected.

jandubois commented on adoc/Release-Notes.adoc:
```diff
@@ -33,6 +33,109 @@ endif::[]
+
+* If you are upgrading {product} to 1.5.2 and use an external database in 1.5.1, please contact SUSE Technical Support on guidance with the help of the backup and restore cli.
+
+Please refer to the external database section of the documentation for more configuration options.
```
Add a link?

gaktive replied:
Based on past practice, this gets tricky based on the various ways to contact support along with the links.

jandubois commented on adoc/Release-Notes.adoc:
```diff
@@ -33,6 +33,109 @@ endif::[]
+* Expanded GoRouter ciphersuite to be compatible with a wider range of clients
+* Made Helm chart compatible with {k8s} 1.16
+* Use system cert store if `UAA_CA_CERT` is not specified
+* Ensure the `HTTP_PORT` is exposed for `file_server` to ensure upgrades do not require longer app downtime
```
This is an internal detail that doesn't need to be listed. It is also not a regression itself, just makes upgrades possible with running apps.

jandubois commented on adoc/Release-Notes.adoc:
```diff
@@ -33,6 +33,109 @@ endif::[]
+** Optimized startup time for various roles
+** Added/fixed podAntiAffinity rules for various roles
+* The Stratos UI has been updated to version 2.7.0:
+** Add support for {caasp} V4.1 and {k8s} 1.16+ environments
```
"Added" because all other bullets are past tense!

Nit-pickers-r-us! :)

gaktive replied:
Added-ed.

gaktive committed:
incorporating jdubois' suggestions

HartS commented on adoc/Release-Notes.adoc:
```diff
@@ -33,20 +33,122 @@ endif::[]
+
+`env.CREDHUB_DB_REQUIRE_TLS` - this is set to `true` by default and is used by credhub to communicate over TLS to the external database. Set it to `false` to disable TLS connections to the external database server from credhub. 
+
+Note that the external database server *must* support both TLS and non-encrypted connections; if the external database server only accepts TLS connections some SCF components will not be able to communicate with the database server.
```
This only applies when TLS is enabled in the UAA + SCF parameters. A DB instance which doesn't support TLS will work when UAA + SCF are configured to use plaintext communication

gaktive replied:
Sentence revised to indicate that it applies when TLS enabled.

HartS commented on adoc/Release-Notes.adoc:
```diff
@@ -33,20 +33,122 @@ endif::[]
+
+* If you are upgrading {product} to 1.5.2 and already use Minibroker to connect to external databases and are using {k8s} 1.16 or higher, which is the case with {caasp} 4.1, you will need to update the database version to a compatible version and migrate your data over via the database's suggested mechanism. This may require a database export/import.
+
+* Starting with {product} 1.5.2, you no longer need to set `UAA_CA_CERT` when using an external UAA with a certificate signed by a well known Certificate Authority (CA). It is only needed when you use an external UAA with either a certificate generated by the `secret-generator` or a self-signed certificate.
```
This may also apply to embedded UAA. When `UAA_SERVER_CERT` and `UAA_SERVER_CERT_KEY` values are provided (see https://documentation.suse.com/suse-cap/1.5.1/single-html/cap-guides/#sec-cap-cert-config)  , the same is true; if the UAA cert is signed by a well-known authority, `UAA_CA_CERT` doesn't need to be provided, and if it's not, `UAA_CA_CERT` does need to be provided

gaktive replied:
How sure are you based on the word _may_? Whatever we put here should be factual and tested, though this can be updated post-release if we need to validate.

gaktive replied:
Leaving as is for now.

gaktive committed:
clarified external db with TLS communications

gaktive committed:
refining TLS & external DB note

nwmac commented on adoc/Release-Notes.adoc:
```diff
@@ -33,20 +33,122 @@ endif::[]
+** Optimized startup time for various roles
+** Added/fixed podAntiAffinity rules for various roles
+* The Stratos UI has been updated to version 2.7.0:
+** Added support for {caasp} V4.1 and {k8s} 1.16+ environments
```
Stratos already supports Kube 1.16+ prior to 2.7.0 - suggest you replace content with:

* The Stratos UI has been updated to version 2.7.0:
** Improvements to the Kubernetes Dashboard integration (improved configuration and reliability)
** Improved support for synchronous creation of service instances
** Branding updated to "SUSE Stratos Console"

gaktive replied:
Thank you sir for the corrected info.

gaktive committed:
incorporating stratos suggestions by nwmac

troytop commented on adoc/Release-Notes.adoc:
```diff
@@ -33,20 +33,123 @@ endif::[]
+** cf-deployment has been updated to 12.17
+** Optimized startup time for various roles
+** Added/fixed podAntiAffinity rules for various roles
+* The Stratos UI has been updated to version 2.7.0:
```
Just "Stratos Console" (no "the")

gaktive replied:
Done.

troytop commented on adoc/Release-Notes.adoc:
```diff
@@ -33,20 +33,123 @@ endif::[]
+** Improved support for synchronous creation of service instances
+** Branding updated to "SUSE Stratos Console"
+** For a full list of features and fixes see https://github.com/SUSE/stratos/blob/v2-master/CHANGELOG.md#270.
+* The Stratos Metrics has been updated to version 1.1.2:
```
Just "Stratos Metrics" (no "the")

troytop commented on adoc/Release-Notes.adoc:
```diff
@@ -33,20 +33,123 @@ endif::[]
 ==== What Is New?
 * {scf} has been updated to version 2.19.1:
-** Support for Eirini SSH feature
+** for Eirini SSH feature
```
"SSH to application instances in Eirini"

gaktive replied:
Done

gaktive committed:
incorporating troytop's suggestions
